python/feldera/rest/client.py

```
@@ -0,0 +1,455 @@
+            version=descriptor.get("version"),
+            description=descriptor.get("description"),
+            id=descriptor.get("id"),
+            config=descriptor.get("config"),
+            state=pipeline.get("state"),
+            attached_connectors=[
+                AttachedConnector(
+                    connector_name=con.get("connector_name"),
+                    is_input=con.get("is_input"),
+                    relation_name=con.get("relation_name"),
+                    name=con.get("name"),
+                )
+                for con in attached_connectors
+            ] if attached_connectors else []
+        )
+
+    def pipelines(self) -> list[Pipeline]:
+        """
+        Get all pipelines
+        """
+
+        resp = self.http.get(
+            path="/pipelines",
+        )
+
+        return [self.__pipeline_from_dict(pipeline) for pipeline in resp]
+
+    def get_pipeline(self, name: str) -> Pipeline:
+        """
+        Get a pipeline by name
+        """
+
+        resp = self.http.get(
+            path=f"/pipelines/{name}",
+        )
+
+        return self.__pipeline_from_dict(resp)
+
+    def create_pipeline(self, pipeline: Pipeline):
+        """
+        Create a pipeline
+        """
+
+        body = {
+            "config": pipeline.config,
+            "description": pipeline.description or "",
+            "connectors": pipeline.attached_connectors,
+            "program_name": pipeline.program_name,
+        }
+
+        resp = self.http.put(
+            path=f"/pipelines/{pipeline.name}",
+            body=body,
+        )
+
+        pipeline.id = resp.get("pipeline_id")
+
+    def get_pipeline_config(self, name: str) -> dict:
+        """
+        Get the configuration of a pipeline by name
+        """
+
+        resp = self.http.get(
+            path=f"/pipelines/{name}/config",
+        )
+
+        return resp
+
+    def validate_pipeline(self, name: str) -> bool:

```

We probably want to return an error description if it's not valid.